src/counts.cpp

```diff
@@ -0,0 +1,82 @@
+/// @file
+/// SPDX-License-Identifier: GPL-3.0-or-later
+/// @author Simon Heybrock
+/// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+/// National Laboratory, and European Spallation Source ERIC.
+#include "counts.h"
+#include "dataset.h"
+#include "except.h"
+
+namespace counts {
+
+auto getBinWidths(const Dataset &d, const std::initializer_list<Dim> &dims) {
+  std::vector<Variable> binWidths;
+  for (const auto dim : dims) {
+    const auto &coord = d(dimensionCoord(dim));
```

Comment:
Haven't tried, but what happens if there is no matching `Coord` var in the dataset... we get a sensible error message?